### PR TITLE
Added plan_only flags to pick and place in Kinetic

### PR DESCRIPTION
**Pull request by DavidWatkins**
### Description

The pick/place functions in moveit_commander did not have a way to only plan the action. I have added optional parameters onto the pick/place calls in moveit_commander which is defaulted to false to ensure no changes in existing code will be required. 

### Checklist
- [X] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Optional: Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Optional: Decide if this should be cherry-picked to other current ROS branches (Indigo, Jade, Kinetic)

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning_interface/move_group_interface/src/demo.cpp [hunk 1]
```diff
@@ -68,7 +68,7 @@ void demoPick(moveit::planning_interface::MoveGroupInterface& group)
 
     grasps.push_back(g);
   }
-  group.pick("bubu", grasps);
+  group.pick("bubu", grasps, false);
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/src/demo.cpp]: for readability can you make this a const?

    const bool plan_only = false;
    (..., plan_only)

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp [hunk 2]
```diff
@@ -795,7 +797,7 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
       return MoveItErrorCode(moveit_msgs::MoveItErrorCodes::FAILURE);
     }
 
-    return pick(object.id, response.grasps);
+    return pick(object.id, response.grasps, false);
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp]: same here for const

File: moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_manipulation.cpp [hunk 3]
```diff
@@ -343,15 +343,15 @@ void MotionPlanningFrame::pickObject()
   {
     move_group_->setSupportSurfaceName(support_surface_name_);
   }
-  if (move_group_->pick(pick_object_name_[group_name]))
+  if (move_group_->pick(pick_object_name_[group_name], false))
```
Comment (davetcoleman) [inline moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_manipulation.cpp]: same here

**Comment by DavidWatkins** [comment]
I added the change to the arguments so that they are constant. Let me know if that doesn't line up with what you meant.

File: moveit_ros/planning_interface/move_group_interface/src/demo.cpp (hunk 1 above)
Comment (davetcoleman) [reply moveit_ros/planning_interface/move_group_interface/src/demo.cpp]: I had actually meant to just change this line (and other similar lines) to not have a hidden false value but instead make it a const right before the call:

    const bool plan_only = false;
    group.pick("bubu", grasps, plan_only);

Comment (davetcoleman) [reply moveit_ros/planning_interface/move_group_interface/src/demo.cpp]: But the approach you used of having default values in the function headers i actually like better, except please remove the ``const``

File: moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h [hunk 4]
```diff
@@ -754,15 +754,15 @@ class MoveGroupInterface
   /** \brief Pick up an object
 
       This applies a number of hard-coded default grasps */
-  MoveItErrorCode pick(const std::string& object);
+  MoveItErrorCode pick(const std::string& object, const bool plan_only=false);
```
Comment (davetcoleman) [inline moveit_ros/planning_interface/move_group_interface/include/moveit/move_group_interface/move_group_interface.h]: please remove the const here and throughout

**Comment by DavidWatkins** [comment]
Thanks for the reminder. Updated it again.

**Comment by mcevoyandy** [comment]
@davetcoleman looks like @DavidWatkins addressed all your feedback. Anything else stopping this PR from moving forward?

**Comment by davetcoleman** [comment]
@mcevoyandy this PR needs a second review, do you +1 it?

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp [hunk 5]
```diff
@@ -1522,21 +1524,21 @@ moveit::planning_interface::MoveItErrorCode moveit::planning_interface::MoveGrou
...
 
 moveit::planning_interface::MoveItErrorCode moveit::planning_interface::MoveGroupInterface::pick(
-    const std::string& object, const std::vector<moveit_msgs::Grasp>& grasps)
+    const std::string& object, const std::vector<moveit_msgs::Grasp>& grasps, bool plan_only)
 {
-  return impl_->pick(object, grasps);
+  return impl_->pick(object, grasps, plan_only);
 }
 
 moveit::planning_interface::MoveItErrorCode
```
Comment (v4hn) [inline moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp]: The `plan_only` flag should also be added to the `planGraspsAndPick` methods below.

File: moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_manipulation.cpp [hunk 6]
```diff
@@ -351,7 +351,7 @@ void MotionPlanningFrame::pickObject()
 
 void MotionPlanningFrame::placeObject()
 {
-  move_group_->place(place_object_name_, place_poses_);
+  move_group_->place(place_object_name_, place_poses_, false);
```
Comment (v4hn) [inline moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_manipulation.cpp]: Please remove this again, it's not required.

File: moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp (hunk 5 above)
Comment (DavidWatkins) [reply moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp]: Added the flag to `planGraspsAndPick` too

File: moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_manipulation.cpp (hunk 6 above)
Comment (DavidWatkins) [reply moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_manipulation.cpp]: Removed

**Comment by DavidWatkins** [comment]
I've addressed the new concerns, any additional thoughts?

File: moveit_commander/src/moveit_commander/move_group.py [hunk 7]
```diff
@@ -67,6 +67,10 @@ def get_joints(self):
         """ Get the joints of this group """
         return self._g.get_joints()
 
+    def get_named_targets(self):
+        """ Get the named targets available in MoveIt! """
+        return self._g.get_named_targets()
+
```
[the remark this answers is not in the document]
Comment (DavidWatkins) [reply moveit_commander/src/moveit_commander/move_group.py]: Removed

**Comment by v4hn** [comment]
Thank you for the swift cleanup!

Merged and cherry-picked to melodic.

**Pull request merged**